srcpkgs/vosk-api/template

```diff
@@ -0,0 +1,78 @@
+# Template file for 'vosk-api'
+pkgname=vosk-api
+version=0.3.43
+revision=1
+create_wrksrc=yes
+build_wrksrc="vosk-api-${version}/src"
+_kaldi_commit=ecb4b47159501ef0d67888633e65b8fc0c1f79d1
+build_style=gnu-makefile
+make_use_env=yes
+make_build_args="KALDI_ROOT=../../kaldi-${_kaldi_commit} HAVE_OPENBLAS_NO_F2C=1 HAVE_OPENBLAS_CLAPACK=0"
+makedepends="blas-devel lapack-devel openblas-devel libgomp-devel openfst-devel"
+short_desc="Offline speech recognition toolkit"
+maintainer="John Gebbie <me@johngebbie.com>"
+license="Apache-2.0"
+homepage="https://alphacephei.com/vosk"
+distfiles="https://github.com/alphacep/vosk-api/archive/v${version}/vosk-api-${version}.tar.gz
+https://github.com/alphacep/kaldi/archive/${_kaldi_commit}.tar.gz
+https://alphacephei.com/vosk/models/vosk-model-small-en-us-0.15.zip"
+checksum="50fa82bfbfe3fc5be868237c84ac180eb55da0b65b1803bf022899673afda024
+39cadd5a351a5513e3e4e1f71ac4b4f018a592a856a55f1d32c159e22e799f64
+30f26242c4eb449f948e42cb302dd7a686cb29a3423a8367f99ff41780942498"
+shlib_provides="libvosk.so"
+
+do_patch() {
+	# https://github.com/alphacep/kaldi/pull/5
+	patch -Np1 -d "${wrksrc}/kaldi-${_kaldi_commit}" < "${PATCHESDIR}/kaldi-lapack.patch"
+	patch -Np1 -d "${wrksrc}/kaldi-${_kaldi_commit}" < "${PATCHESDIR}/kaldi-to-openblas-0.3.21.patch"
+	patch -Np1 -d "${wrksrc}/kaldi-${_kaldi_commit}" < "${PATCHESDIR}/kaldi-to-openfst-1.8.2.patch"
+	# https://github.com/alphacep/vosk-api/pull/1215
+	patch -Np1 -d "${wrksrc}/vosk-api-${version}" < "${PATCHESDIR}/vosk-lapack.patch"
+}
```

Comment:
This can be replaced by `patch_args="-Np1 -d ${pkgname}-${version}"` and `patches/kaldi-*.patch.args` files containing `-Np1 -d kaldi`. The kaldi directory currently contains version in its name like I suggest in a comment in `post_extract`. Then patches/series can be removed.

Not overriding do_patch is more consistent with other packages and gets rid of one unnecessary step when adding a patch.
Also a nitpick that's not important if we modify it this way, we usually try to follow keep the order of build steps in sync with order of functions, patch runs after extract, so do_patch should be after do_extract.

Reply:
Thank you! that is better.